src/Compute/Compute/VirtualMachine/Config/AddAzureVMNetworkInterfaceCommand.cs

```diff
@@ -148,12 +162,14 @@ public override void ExecuteCmdlet()
                             new NetworkInterfaceReference
                             {
                                 Id = nic.Id,
-                                Primary = nic.Primary
+                                Primary = nic.Primary,
+                                DeleteOption = this.IsParameterBound(c => c.DeleteOption) ? this.DeleteOption : null
                             });
                     }
                     else
                     {
                         existingNic.Primary = nic.Primary;
+                        existingNic.DeleteOption = this.IsParameterBound(c => c.DeleteOption) ? this.DeleteOption : null;
```

Comment:
Shouldn't the ternary expression be

existingNic.DeleteOption = this.IsParameterBound(c => c.DeleteOption) ? this.DeleteOption : nic.DeleteOption